crates/router/src/bin/scheduler.rs

```diff
@@ -68,6 +71,16 @@ async fn main() -> CustomResult<(), ProcessTrackerError> {
         [router_env::service_name!()],
     );
 
+    #[allow(clippy::expect_used)]
+    let web_server = Box::pin(start_web_server(
+        state.clone(),
+        scheduler_flow_str.to_string(),
+    ))
+    .await
+    .expect("Failed to create the server");
+
+    tokio::spawn(web_server);
```

Comment:
Can we have something like
```suggestion
    tokio::spawn(move async {
    	web_server.await;
    	logger::error!("The health check probe stopped working!");
    });
```